workflow/controller/dag.go

```diff
@@ -359,18 +350,15 @@ func (woc *wfOperationCtx) executeDAG(ctx context.Context, nodeName string, tmpl
 		err := c.Save(ctx, node.MemoizationStatus.Key, node.ID, node.Outputs)
 		if err != nil {
 			woc.log.WithFields(log.Fields{"nodeID": node.ID}).WithError(err).Error("Failed to save node outputs to cache")
-			node.Phase = wfv1.NodeError
+			woc.markNodeError(nodeName, err)
 		}
 	}
 
-	err = woc.updateOutboundNodesForTargetTasks(dagCtx, targetTasks, nodeName)
-	if err != nil {
-		return nil, err
-	}
+	woc.updateOutboundNodesForTargetTasks(dagCtx, targetTasks, node)
 	return woc.markNodePhase(nodeName, wfv1.NodeSucceeded), nil
 }
 
-func (woc *wfOperationCtx) updateOutboundNodesForTargetTasks(dagCtx *dagContext, targetTasks []string, nodeName string) error {
+func (woc *wfOperationCtx) updateOutboundNodesForTargetTasks(dagCtx *dagContext, targetTasks []string, node ) {
```

Comment:
woops, I missed the type here; I thought there was a reason I didn't push this up for a while, I just forgot what that was 😅

Reply:
fixed